File: tubepilotaireact.client/src/types/project.ts
export type ProjectStatus = 'Draft' | 'Active' | 'OnHold' | 'Completed' | 'Archived' | 0 | 1 | 2 | 3 | 4;

export type ProjectPriority = 'Low' | 'Medium' | 'High' | 'Critical' | 0 | 1 | 2 | 3;

export interface Project {
  id: string;
  name: string;
  description?: string | null;
  ownerName: string;
  status: ProjectStatus;
  priority: ProjectPriority;
  startDate?: string | null;
  dueDate?: string | null;
  budget: number;
  tags: string[];
  createdAtUtc: string;
  updatedAtUtc?: string | null;
}

export interface ProjectPayload {
  name: string;
  description?: string | null;
  ownerName: string;
  status: ProjectStatus;
  priority: ProjectPriority;
  startDate?: string | null;
  dueDate?: string | null;
  budget: number;
  tags: string[];
}

export interface ProjectFilters {
  search?: string;
  status?: string;
  priority?: string;
}

export interface ProjectQuery extends ProjectFilters {}

File: TubePilotAI.Domain/Entities/Project.cs
using TubePilotAI.Domain.Enums;

namespace TubePilotAI.Domain.Entities;

public sealed class Project
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public string? Description { get; set; }

    public string OwnerName { get; set; } = string.Empty;

    public ProjectStatus Status { get; set; } = ProjectStatus.Draft;

    public ProjectPriority Priority { get; set; } = ProjectPriority.Medium;

    public DateOnly? StartDate { get; set; }

    public DateOnly? DueDate { get; set; }

    public decimal Budget { get; set; }

    public string TagsJson { get; set; } = "[]";

    public DateTime CreatedAtUtc { get; set; }

    public DateTime? UpdatedAtUtc { get; set; }

    public ICollection<GeneratedContent> GeneratedContents { get; set; } = new List<GeneratedContent>();
}
